charmhelpers/contrib/openstack/context.py

```diff
@@ -1128,7 +1131,8 @@ class ApacheSSLContext(OSContextGenerator):
     user = group = 'root'
 
     def enable_modules(self):
-        cmd = ['a2enmod', 'ssl', 'proxy', 'proxy_http', 'headers']
+        # NOTE(seyeongkim) : remoteip is for proxy protocol v2
```

Comment:
should adding this new module be conditional on if config('haproxy-enable-proxyv2')?

Reply:
Right. I considered it in the beginning but like ssl module, I just put 'enabling remoteip module' only, Because we can enable/disable it by configuration option.
How do you think about this approach? conditional enabling could be better? 
Thanks so much.

```
    def enable_modules(self):
        cmd = ['a2enmod', 'ssl', 'proxy', 'proxy_http', 'headers']
        check_call(cmd)
        # remoteip is only needed for proxy protocol v2
        if config('haproxy-enable-proxyv2'):
            check_call(['a2enmod', 'remoteip'])
        else:
            # Disable remoteip module
            try:
                check_call(['a2dismod', 'remoteip'])
            except CalledProcessError:
                # ignore error
                pass
```